pkg/system/meminfo_deprecated.go

```diff
@@ -1,16 +1,12 @@
 package system
 
-import "github.com/docker/docker/pkg/sysinfo"
-
 // MemInfo contains memory statistics of the host system.
 //
 // Deprecated: use [sysinfo.Memory].
-type MemInfo = sysinfo.Memory
+type MemInfo = struct{}
```

Comment:
Can we just remove this function, and add "Memory was previously available as [system.MemInfo]" to the comment lines of `sysinfo.Memory`?

Reply:
Yeah, I was really on the fence to either remove it altogether, but then consumers would have no guidance at all where it went.


To be fair; the list of consumers is very limited; https://grep.app/search?q=.ReadMemInfo%28%29&filter[lang][0]=Go

(ignoring the `vendor/xx` and PodMan code (which uses its own fork of this), I _think_ it's only `nerdctl` and `amazon-ecs-agent`;

- https://github.com/containerd/nerdctl/blob/5da1481d3342d75bf356469afeac2fb534363047/pkg/infoutil/infoutil_linux.go#L123
- https://github.com/aws/amazon-ecs-agent/blob/13d1e854decd15a8839ea76cc1cd1b73b9f9d03b/agent/api/ecsclient/client.go#L310

Reply:
@corhere @cpuguy83 any thoughts? Do you think we should remove it altogether, or "some attempt on creating stub (but compile-time error)?

Reply:
> but then consumers would have no guidance at all where it went.

I expect them to run `git grep MemInfo` and find the guidance comment (`Memory was previously available as [system.MemInfo]`).

Reply:
I prefer option C: solve the dependency-graph problem without breaking any users. Move the `MemInfo` and `ReadMemInfo` implementations into a new `pkg/meminfo` package and update both `pkg/system` and `pkg/sysinfo` to have deprecated aliases.

Reply:
I'm a bit on the fence doing so. These functions were dropped in `pkg/system` as it was already a bit of a grab-bag of functions, but were only actually meant for internal use (most notably, together with `sysinfo`, which provided the other info about the system.

If we _would_, I would not update the aliases in `pkg/sysinfo`, as that location has never been released (only on master).